### PR TITLE
DAOS-8334 dfs: add dfs_cont_create_with_label

### PR DESCRIPTION
**Pull request by mchaarawi**
- convenience function similar to daos_cont_create_with_label so user
doesn't need to set set label property and can just pass the label
through API.
- change go cont create to let daos set cont uuid if user does not specify

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/dfs/dfs.c [hunk 1]
```diff
@@ -1476,6 +1480,52 @@ dfs_cont_create2(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr, daos_handle
...
+	int			rc;
+
+	label_prop = daos_prop_alloc(1);
+	if (label_prop == NULL)
+		return ENOMEM;
+
+	label_prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+	rc = daos_prop_entry_set_str(&label_prop->dpp_entries[0], label, DAOS_PROP_LABEL_MAX_LEN);
+	if (rc)
+		D_GOTO(out_prop, rc);
```
Comment (ashleypittman) [inline src/client/dfs/dfs.c:1500]: This will be a daos error number I assume?

File: src/client/dfs/dfs.c [hunk 2]
```diff
@@ -1476,6 +1480,52 @@ dfs_cont_create2(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr, daos_handle
...
+		orig = attr->da_props;
+		attr->da_props = merged_props;
+	} else {
+		attr->da_props = label_prop;
+	}
+
+	if (cuuid == NULL) {
+		uuid_t u;
+
+		rc = dfs_cont_create_int(poh, &u, false, NULL, attr, coh, dfs);
```
Comment (mjmac) [inline src/client/dfs/dfs.c:1518]: Seems like the uuid is thrown away here, is that intentional?

**Review by mjmac — APPROVED** [review]
The changes in container.go look good to me... The label/uuid stuff turned into a bit of spaghetti, this patch makes it better.

File: src/client/dfs/dfs.c (hunk 2 above)
Comment (mchaarawi) [reply src/client/dfs/dfs.c:1518]: yes, this is when a user creates with label, but does not pass a sink buffer for the uuid, as they don't care about using it.

**Comment by mchaarawi** [comment]
lost a race with @ashleypittman printf change

File: src/control/cmd/daos/container.go [hunk 3]
```diff
@@ -297,6 +296,15 @@ func (cmd *containerCreateCmd) Execute(_ []string) (err error) {
...
+
+	cmd.contUUID, err = uuidFromC(ap.c_uuid)
+	if err != nil {
+		return err
+	}
+	if cmd.contUUID == uuid.Nil {
+		cmd.contLabel = C.GoString(&ap.cont_str[0])
+	}
+
 	cmd.log.Debugf("created container: %s", cmd.contUUID)
```
Comment (kccain) [inline src/control/cmd/daos/container.go:308]: In a scenario where a path is given, cont_create_uns_hdlr() is run above, this line will produce this output
created container: 00000000-0000-0000-0000-000000000000

Suggest above where cmd.contLabel is assigned to add cmd.log.Debugf("created container: %s", cmd.contLabel), and make this statement be in the else of the conditional.

**Review by kccain — CHANGES_REQUESTED** [review]

File: src/control/cmd/daos/container.go (hunk 3 above)
Comment (mchaarawi) [reply src/control/cmd/daos/container.go]: fixed

**Review by kccain — APPROVED** [review]

File: src/control/cmd/daos/container.go [hunk 4]
```diff
@@ -301,29 +301,31 @@ func (cmd *containerCreateCmd) Execute(_ []string) (err error) {
 	if err != nil {
 		return err
 	}
+
+	var co_id string
```
Comment (mjmac) [inline src/control/cmd/daos/container.go:305]: nit: snake_case variable names will cause linter warnings for Go developers. Can be fixed later though.

**Review by mjmac — APPROVED** [review]

**Pull request merged**